gym/wrappers/frame_stack.py

```diff
@@ -0,0 +1,98 @@
+from collections import deque
+import numpy as np
+
+from gym.spaces import Box
+from gym import ObservationWrapper
+
+
+class LazyFrames(object):
+    r"""Ensures common frames are only stored once to optimize memory use. 
+
+    To further reduce the memory use, it is optionally to turn on lz4 to 
+    compress the observations.
+
+    .. note::
+
+        This object should only be converted to numpy array just before forward pass. 
+
+    """
+    def __init__(self, frames, lz4_compress=False):
+        from lz4.block import compress
+        from lz4.block import decompress
+        if lz4_compress:
+            self.shape = frames[0].shape
+            frames = [compress(frame) for frame in frames]
+        self._frames = frames
+        self.lz4_compress = lz4_compress
+
+    def __array__(self, dtype=None):
+        if self.lz4_compress:
+            frames = [np.frombuffer(decompress(frame), dtype=np.uint8).reshape(self.shape) for frame in self._frames]
```

Comment:
```suggestion
            frames = [
                np.frombuffer(decompress(frame), dtype=np.uint8).reshape(self.shape)
                for frame in self._frames
            ]
```

Reply:
Are these frames always images? Should the `dtype` be something else than `uint8`?